File: 3-Big_O/big_o/lib/show_all_pairs.ex
defmodule ShowAllPairs do
  def show(lst) do
    for i <- lst do
      for j <- lst do
        IO.puts("#{i} #{j}")
      end
    end
  end
end

# => O(n^2)
# ShowAllPairs.show(["a", "b", "c", "d", "e"])

File: 3-Big_O/show_all_pairs.rb
# frozen_string_literal: true

def show_all_pairs(arr)
  arr.each do |i|
    arr.each do |j|
      puts "#{i} #{j}"
    end
  end
end

boxes = %w[a b c d e]
show_all_pairs(boxes) # => O(n^2)
